backends/github/labels/github-labels.js

```diff
@@ -0,0 +1,165 @@
+import GithubAPI from "../api/github-api.js";
+
+/**
+ * Backend for performing operations with labels on GitHub repositories.
+ * @category GitHub
+ */
+export default class GithubLabels extends GithubAPI {
+	// Do we need this at all?
+	static urlsKnown = [
+		{hostname: "api.github.com", pathname: "repos/:owner/:repo/labels"},
+		{hostname: "github.com", pathname: "/:owner/:repo/labels"},
+	];
+
+	static phrases = {
+		not_authenticated: "Please log in to perform actions with labels on GitHub.",
+		no_labels: "There are no labels to work with. Use the delete() method if you were to delete all existing labels in one go.",
+		success: (action) => `Labels ${action}d:`,
+		failure: (action) => `Labels failed to ${action}:`,
+	};
+
+	/**
+	 * Performs operations (create, update, delete) with labels.
+	 * @param {Array<any>} data Labels to create, update, or delete.
+	 * @returns {Promise<Object>} Promise that is resolved with an object based on the results of performed operations.
+	 */
+	async put (data, {ref = this.ref, force = false, skipDeletion = false} = {}) {
+		if (!this.isAuthenticated()) {
+			console.warn(this.constructor.phrase("not_authenticated"));
+			return null;
+		}
+
+		if (!data?.length && !force) {
+			// We don't want to delete all existing labels accidentally
+			console.warn(this.constructor.phrase("no_labels"));
+			return null;
+		}
+
+		if (!this.data) {
+			// Existing labels are not yet fetched. Fetching...
+			try {
+				await this.load(this.source);
+			}
+			catch (err) {
+				if (err.status === 404) {
+					return null;
+				}
+			}
+		}
+
+		let ret = {};
+		let toCreate = data.filter(label => !this.data.find(l => l.name === label.name));
```

Comment:
This will fail if I’ve renamed the label, which is precisely what I want to be doing. These labels have an id, better use that to look them up.